### PR TITLE
refactor: signOutOnNetworkChange handler on mapToSIWX

### PR DESCRIPTION
**Pull request by enesozturk**
# Description

Refactor `mapToSIWX` function to handle network changes depending on the `signOutOnNetworkChange` flag as it shouldn't ask for signature if it's `false`.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/siwe/src/mapToSIWX.ts [hunk 1]
```diff
@@ -183,15 +183,22 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
...
+
+        const lowercaseAddress = address?.toLowerCase()
+        const caipAddress = `${chainId}:${address}`
+
+        const signOutOnNetworkChange = siwe.options.signOutOnNetworkChange
+
+        const shouldSignOut = signOutOnNetworkChange
+          ? caipSessionAddress !== caipAddress
+          : lowercaseSessionAddress !== lowercaseAddress
+        console.log('caipSessionAddress', signOutOnNetworkChange)
```
Comment (magiziz) [inline packages/siwe/src/mapToSIWX.ts:199]: Remove ?

File: packages/siwe/src/mapToSIWX.ts [hunk 2]
```diff
@@ -183,15 +183,22 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
...
+        const caipSessionAddress = `${siweCaipNetworkId}:${siweSession?.address}`
+
+        const lowercaseAddress = address?.toLowerCase()
+        const caipAddress = `${chainId}:${address}`
+
+        const signOutOnNetworkChange = siwe.options.signOutOnNetworkChange
+
+        const shouldSignOut = signOutOnNetworkChange
+          ? caipSessionAddress !== caipAddress
+          : lowercaseSessionAddress !== lowercaseAddress
```
Comment (magiziz) [inline packages/siwe/src/mapToSIWX.ts:198]: We can use this function here ? https://github.com/reown-com/appkit/blob/661548e3b7e2b8650935a3af22303ae3cf685111/packages/appkit-utils/src/HelpersUtil.ts#L19

File: packages/siwe/src/mapToSIWX.ts (hunk 1 above)
Comment (enesozturk) [reply packages/siwe/src/mapToSIWX.ts:199]: Yes?

Comment (enesozturk) [reply packages/siwe/src/mapToSIWX.ts:199]: lol

File: packages/siwe/src/mapToSIWX.ts (hunk 2 above)
Comment (tomiir) [inline packages/siwe/src/mapToSIWX.ts:198]: why does it matter if the flag is set to check for lowecase???

**Review by tomiir — APPROVED** [review]

**Pull request merged**